### ISSUE TITLE
Request - Don't convert nested blocks

### ISSUE THREAD

**Issue by JFaulk1434** [luisalvesmartins/Markdown-Mermaid-Expand-to-PNG#4]
# Request to skip nested code blocks

I was making a guide on mermaid in markdown and came across this extension which is amazing. But one thing I noticed is it also converts nested blocks. See examples below:

**Will be converted:**

````markdown
```mermaid
sequenceDiagram
  Alice->>Bob: Hello Bob, how are you?
  Bob-->>John: How about you, John?
  John-->>Alice: I am good, thanks!
```
````

**Will also be converted but shouldn't:**

`````markdown
````markdown
```mermaid
sequenceDiagram
  Alice->>Bob: Hello Bob, how are you?
  Bob-->>John: How about you, John?
  John-->>Alice: I am good, thanks!
```
````
`````
The reason for this is to show mermaid code in markdown along with an example of the rendered image.
End result should look like this:
<img width="711" alt="Screenshot 2023-08-22 at 11 38 17 AM" src="https://github.com/luisalvesmartins/Markdown-Mermaid-Expand-to-PNG/assets/104535802/095cc33c-6fb1-4204-a7ce-0f68985d3965">

**Comment by luisalvesmartins** [comment]
Good catch. I'm not doing a parse of the text, I'm doing a "blind" search. As this can be an edge case I would propose to avoid having the mermaid declaration in all pictures. Otherwise, this is probably a lot of work and I will put this as "feature request" for a future version.
I would be very happy if you want to contribute...
